File: app_flutter_horario_mobile/lib/domain/coordenador.dart
import 'package:app_flutter_horario_mobile/domain/curso.dart';
import 'package:app_flutter_horario_mobile/domain/profissional.dart';

class Coordenador extends Profissional {
  List<Curso>? _cursos;

  Coordenador({List<Curso>? cursos});

  get cursos => this._cursos;

  set cursos(value) => this._cursos = value;
}

File: backend/systempus/src/main/java/br/com/systempus/systempus/domain/Coordenador.java
package br.com.systempus.systempus.domain;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonManagedReference;

import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;
import jakarta.persistence.PreRemove;
import jakarta.persistence.Table;
import jakarta.persistence.DiscriminatorValue;

@Entity
@Table(name = "coordenador")
@DiscriminatorValue("1")
public class Coordenador extends Profissional{

    @OneToMany(mappedBy = "coordenador"/* , cascade = CascadeType.ALL*/)
    // @JsonManagedReference(value = "curso_coordenador")
    private List<Curso> cursos;

    public void setCursos(List<Curso> cursos){
        this.cursos = cursos;
    }

    public List<Curso> getCursos(){
        return cursos;
    }

    @PreRemove
    private void preRemove() {
        for (Curso curso : cursos) {
            curso.setCoordenador(null);
        }
    }
}